FileShareClient/FileShareApiClient.cs

```diff
@@ -99,6 +100,40 @@ public async Task<BatchSearchResponse> Search(string searchQuery, int? pageSize
             }
         }
 
+        public async Task<IResult<BatchSearchResponse>> Search(string searchQuery, CancellationToken cancellationToken, int? pageSize = null, int? start = null)
```

Comment:
There is much duplicated code in the two Search methods. Could you please move the duplicated code to a new private method and call it from the Search methods?
I have created gist to explain what I mean: https://gist.github.com/crdandoukho/dbc90d4389f3afd27fc36c58bf6e3b55

Reply:
resolve